Corona-Warn-App/src/main/java/de/rki/coronawarnapp/covidcertificate/validation/core/business/BusinessValidator.kt

```diff
@@ -0,0 +1,61 @@
+package de.rki.coronawarnapp.covidcertificate.validation.core.business
+
+import dagger.Reusable
+import de.rki.coronawarnapp.covidcertificate.common.certificate.DccData
+import de.rki.coronawarnapp.covidcertificate.common.certificate.DccJsonSchema
+import de.rki.coronawarnapp.covidcertificate.common.certificate.DccV1
+import de.rki.coronawarnapp.covidcertificate.validation.core.DccValidationRepository
+import de.rki.coronawarnapp.covidcertificate.validation.core.business.wrapper.CertLogicEngineWrapper
+import de.rki.coronawarnapp.covidcertificate.validation.core.business.wrapper.filterRelevantRules
+import de.rki.coronawarnapp.covidcertificate.validation.core.business.wrapper.type
+import de.rki.coronawarnapp.covidcertificate.validation.core.country.DccCountry
+import org.joda.time.Instant
+import javax.inject.Inject
+
+@Reusable
+class BusinessValidator @Inject constructor(
+    private val certLogicEngineWrapper: CertLogicEngineWrapper,
+    private val ruleRepository: DccValidationRepository,
+    private val dccJsonSchema: DccJsonSchema,
+) {
+    suspend fun validate(
+        arrivalCountry: DccCountry,
+        validationClock: Instant,
+        certificate: DccData<out DccV1.MetaData>,
+    ): BusinessValidation {
+
+        // TODO update value sets
+
+        ruleRepository.refresh()
```

Comment:
I think this can be removed.
The validator works with the data that is currently available.
The refresh of countries and rules should happen in the UI, before the user is allowed to reach the screen where date and country is selected, any error during refresh at that point will prevent the user from continuing.

See:
https://github.com/corona-warn-app/cwa-app-android/pull/3588/files#diff-d8aecc75f9c23ee2a9d44a769962d8c734ba5735cb76923ece40fa0992a0eae2R117-R126

Reply:
Removed in next PR https://github.com/corona-warn-app/cwa-app-android/pull/3663/files#diff-7bb7162fbf68792f8a6ebdc2dfe685c21675e1702a085d7cbe39a6ec43161ebbL26